File: primary-algorithm/src/main/java/moon/scala/RemoveDuplicates.scala
package moon.scala

class RemoveDuplicates {
  def removeDuplicates(nums:Array[Int]):Int={
    if(nums.length==0){
      return 0
    }
    var i=0
    for(j<-nums.indices){
      if(nums(i)!=nums(j)){
        i+=1
        nums(i)=nums(j)
      }
    }
    i+1
  }
}

File: data-structure/src/main/java/array/RemoveDuplicates.java
package array;

public class RemoveDuplicates {
    public int solve(int[] nums) {
        if(nums.length<=0){
            return 0;
        }
        int length=1,last=nums[0];
        for(int i=1;i<nums.length;i++){
            if(nums[i]==last){
                continue;
            }
            nums[length++]=nums[i];
            last=nums[i];
        }
        return length;
    }
}
